frontend/sideMenu/sideMenu.component.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+(function() {
+    angular
+    .module('app')
+    .component('sideMenu', {
+        templateUrl: "app/sideMenu/sideMenu.html",
+        controller: sideMenuController,
+        controllerAs: "sideMenuCtrl",
+        bindings: {
+            type: '@'
+        }
+    });
+
+    function sideMenuController(AuthService, STATES, $state, $mdSidenav, 
+        $mdDialog, HomeItemsFactory, ManageInstItemsFactory, InstitutionService) {
+        
+        const sideMenuCtrl = this;
+        const HOME_TYPE = "HOME";
+        const MANAG_INST_TYPE = "MANAGE_INSTITUTION";
+        const INST_PAGE_TYPE = "INSTITUTION_PAGE";
+
+        sideMenuCtrl.user = AuthService.getCurrentUser();
+        
+        sideMenuCtrl.$postLink = () => {
+            setup();
+        };
+        
+        const setup = () => {
+            if(sideMenuCtrl.isType(HOME_TYPE)) {
```

Comment:
isnt better to create two functions: setupHomeMenu and setupInstitutionMenu
and this setup function be something like
sideMenuCtrl.isType(HOME_TYPE) && setupHomeMenu
!sideMenuCtrl.isType(HOME_TYPE) && setupInstituionMenu